File: 1233-remove-sub-folders-from-the-filesystem/1233-remove-sub-folders-from-the-filesystem.java
class Solution {
    public List<String> removeSubfolders(String[] folder) {
        Arrays.sort(folder);
        var parents = new ArrayList<String>();

        var parent = "foo";
        for(var path : folder){
            if(path.startsWith(parent + "/")) continue;
            parent = path;
            parents.add(parent);
        }

        return parents;
    }
}

File: 1233-remove-sub-folders-from-the-filesystem/1233-remove-sub-folders-from-the-filesystem.py
class Solution:
    def removeSubfolders(self, folder: List[str]) -> List[str]:
        folder.sort()
        parents = []

        parent = "foo"
        for path in folder:
            if path.startswith(parent + "/"): continue
            parent = path
            parents.append(parent)

        return parents